sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -1370,10 +1370,10 @@ object DecimalAggregates extends Rule[LogicalPlan] {
 }
 
 /**
- * Converts local operations (i.e. ones that don't require data exchange) on LocalRelation to
- * another LocalRelation.
+ * Converts local operations (i.e. ones that don't require data exchange) on [[LocalRelation]] to
+ * another [[LocalRelation]].
  *
- * This is relatively simple as it currently handles only 2 single case: Project and Limit.
+ * This rule currently handles 3 cases: [[Project]], [[Limit]] and [[Filter]].
```

Comment:
Maybe we should just not specify this in the docs. Are you sure the `[[...]]` links work when you generate scaladoc? the test builder doesn't check. I've seen many errors/warnings generated by some types of links.
If so again maybe easier to restrict this to merely removing this line.

Reply:
`catalyst` isn't meant to be a public API (for our convenience, it's not all marked private in the source). That's why it's not in the docs and yeah that's why writing references to it won't work, as I recall.

For this comment I think it's simplest to remove it, or leave it but don't add references, or move it to an inline comment.

Reply:
Thanks a lot @srowen for your explanation. Since it's not a public API and the code is clear, it makes sense to remove this line, this can also eliminate the needs to modify this line when new cases are added. I have removed this line in the new commit.